tests/unit/test_data.py

```diff
@@ -9,3 +15,44 @@ def test_repeating_loader():
         assert next(loader) == 1
         assert next(loader) == 2
         assert next(loader) == 3
+
+
+@pytest.mark.parametrize('train_batch_size, drop_last',
+                         [(1,
+                           True),
+                          (4,
+                           True),
+                          (1,
+                           False),
+                          (4,
+                           False)])
+def test_dataloader_drop_last(tmpdir, train_batch_size, drop_last):
+    config_dict = {
+        "steps_per_print": 1,
+    }
+    args = args_from_dict(tmpdir, config_dict)
+    hidden_dim = 10
+
+    model = SimpleModel(hidden_dim)
+
+    @distributed_test(world_size=[1])
+    def _test_dataloader_drop_last(args, model, hidden_dim):
+        optimizer = torch.optim.AdamW(params=model.parameters())
+        batch_size = model.train_micro_batch_size_per_gpu()
+        model, _, _, _ = deepspeed.initialize(args=args,
+                                              train_batch_size=train_batch_size,
```

Comment:
This will unfortunately break since `train_batch_size` is not an argument of `deepspeed.intialize()`.

Reply:
My bad. Corrected it.